src/launcher.c

```diff
@@ -293,10 +314,29 @@ static void *load_app(char *name)
     }
   }
 
-  if (patch_svc(code, size) != 0) {
+
+
+  uint32_t start = 0;
+  uint32_t code_length = size;
+  if (app->elf.code_start && app->elf.code_end) {
+    start = app->elf.code_start - 0xc0de0000;
+    code_length = app->elf.code_end - app->elf.code_start;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_WRITE) != 0) {
+    warn("mprotect(PROT_READ | PROT_WRITE)");
     goto error;
   }
 
+  if (patch_svc(code + start, code_length) != 0) {
+    goto error;
+  }
+
+  if (mprotect(code, size, PROT_READ | PROT_EXEC) != 0) {
+    warn("mprotect(PROT_READ | PROT_EXEC)");
```

Comment:
Maybe be more explicit in the warning or add a comment about what this code does